src/Unirest/Request.php

```diff
@@ -555,8 +555,13 @@ private static function encodeUrl($url)
             $query = '?' . http_build_query(self::getArrayFromQuerystring($query));
         }
 
-        if ($port && $port[0] !== ':') {
-            $port = ':' . $port;
+        if (null !== $port) {
+            if (!is_string($port)) {
+                $port = strval($port);
+            }
+            if ($port[0] !== ':') {
```

Comment:
As strval() returns the unmodified string if you put a string into it, you might consider simplifying the code to

``` php
if ($port && strval($port)[0] !== ':') {
    $port = ':' . $port;
}
```

In my opinion, this would be easier to read and gave a better impression of what the code is doing.

Also, if $port is `''` (empty string), it will get through the null check and this will call an undefined index on $port. This is just a notice in PHP, however, it won't occur with a weak type check.

Reply:
Just updated it, what do you think about this version ? :)

Reply:
Nice! 👍